src/combinatorics/simple_graph/clique.lean

```diff
@@ -123,6 +137,57 @@ def clique_free (n : ℕ) : Prop := ∀ t, ¬ G.is_n_clique n t
 
 variables {G H}
 
+lemma not_clique_free_of_top_embedding {n : ℕ}
```

Comment:
Are you trying to avoid using choice? because it's weird to have this lemma with a long proof before the definition that would golf it ten fold.

Reply:
Sorry, I'm not sure I follow your suggestion. I see `top_embedding_of_not_clique_free`, but that's the converse.